pkgs/tools/filesystems/ceph/default.nix

```diff
@@ -1,44 +1,82 @@
-{ lib, stdenv, runCommand, fetchurl, fetchpatch
+{ lib
+, stdenv
+, runCommand
+, fetchurl
+, fetchFromGitHub
+
+# Build time
+, cmake
 , ensureNewerSourcesHook
-, cmake, pkg-config
-, which, git
-, boost175, xz
-, libxml2, zlib, lz4
-, openldap, lttng-ust
-, babeltrace, gperf
-, gtest
-, cunit, snappy
+, fmt
+, git
 , makeWrapper
-, leveldb, oath-toolkit
-, libnl, libcap_ng
-, rdkafka
+, pkg-config
+, which
+
+# Tests
 , nixosTests
-, cryptsetup
-, sqlite
-, lua
-, icu
+
+# Runtime dependencies
+, arrow-cpp
+, babeltrace
+, boost179
 , bzip2
+, cryptsetup
+, cimg
+, cunit
 , doxygen
+, gperf
 , graphviz
-, fmt
-, python39
+, gtest
+, icu
+, jsoncpp
+, libcap_ng
+, libnl
+, libxml2
+, lttng-ust
+, lua
+, lz4
+, oath-toolkit
+, openldap
+, python310
+, rdkafka
+, rocksdb
+, snappy
+, sqlite
+, utf8proc
+, zlib
+, zstd
 
 # Optional Dependencies
-, yasm ? null, fcgi ? null, expat ? null
-, curl ? null, fuse ? null
-, libedit ? null, libatomic_ops ? null
+, curl ? null
+, expat ? null
+, fuse ? null
+, libatomic_ops ? null
+, libedit ? null
 , libs3 ? null
+, yasm ? null
 
 # Mallocs
-, jemalloc ? null, gperftools ? null
+, gperftools ? null
+, jemalloc ? null
 
 # Crypto Dependencies
 , cryptopp ? null
-, nss ? null, nspr ? null
+, nspr ? null
+, nss ? null
 
 # Linux Only Dependencies
-, linuxHeaders, util-linux, libuuid, udev, keyutils, rdma-core, rabbitmq-c
-, libaio ? null, libxfs ? null, zfs ? null, liburing ? null
+, linuxHeaders
+, util-linux
+, libuuid
+, udev
+, keyutils
+, rdma-core
+, rabbitmq-c
+, libaio ? null
+, libxfs ? null
+, liburing ? null
+, zfs ? null
```

Comment:
Disabling dependecies by overwriting them with null is an anti pattern

Reply:
Out of scope for this upgrade. I'm not the maintainer, I'm merely making sure we have a supported ceph version in the upcoming release.